### DISCUSSION TITLE
import/no-cycle with TypeScript + ESM

### DISCUSSION THREAD

**Discussion by LukeNotable** [oxc-project/oxc#6906] [Q&A]
Trying out oxlint for the first time in the hope that it can at least give us a fast check on `import/no-cycle`. I have a TypeScript project using ESM, thus files like:

```ts
// foo.ts
import { Bar } from './bar.js';

// bar.ts
import { Foo } from './foo.js';
```

However, the rule fails to flag this as a problem. But if I remove the `.js` extension, the rule behaves correctly. However, the `.js` extension is required under ESM. 

I recall a similar issue with the original `eslint-plugin-import`, which required `eslint-import-resolver-typescript`, and I see there's an `oxc-resolver`, so I wonder if I just need to do some additional configuration to use it.

**Comment by Boshen** [comment]
Are there any tsconfig.json configurations that could tell me this extension usage?

The current resolver does not have `extensionAlias` set.

https://github.com/oxc-project/oxc/blob/6e68c9398b35a95a8f3ba39cf0267126aed5486d/crates/oxc_linter/src/service/runtime.rs#L53-L68

I don't really want to add a resolver configuration to oxlintrc 😅

**Reply by LukeNotable** [reply]
> Are there any tsconfig.json configurations that could tell me this extension usage?

The pertinent part would just be:
```json
  "compilerOptions": {
    "module": "nodenext",
    "moduleResolution": "nodenext",
  },
```

> I don't really want to add a resolver configuration to oxlintrc 😅

Right, ideally the built-in resolver would just do the right thing for TypeScript.
